enterprise/api/v1/views.py

```diff
@@ -1483,3 +1484,56 @@ def update(self, request, *args, **kwargs):
     )
     def destroy(self, request, *args, **kwargs):
         return super().destroy(request, *args, **kwargs)
+
+    @action(methods=['post'], detail=True, url_path='link-user')
+    def link_user(self, request, pk=None):
+        """
+        Post
+        Links user using enterprise_customer_key
+        /enterprise/api/enterprise-customer-invite-key/{enterprise_customer_key}/link-user
+
+        Given a enterprise_customer_key, link user to the appropriate enterprise.
+
+        If the key is not found, returns 400
+        If the ket is not valid, return 422
+        """
+        try:
+            enterprise_customer_key_match = models.EnterpriseCustomerInviteKey.objects.get(uuid=pk)
+        except models.EnterpriseCustomerInviteKey.DoesNotExist:
+            return Response({"error": "Could not find Enterprise Customer Invite Key"}, status=HTTP_400_BAD_REQUEST)
+
+        if not enterprise_customer_key_match.is_valid:
+            return Response(
+                {"detail": "Enterprise customer invite key is not valid"},
+                status=status.HTTP_422_UNPROCESSABLE_ENTITY,
+            )
+
+        enterprise_customer = enterprise_customer_key_match.enterprise_customer
+
+        enterprise_user, created = models.EnterpriseCustomerUser.objects.get_or_create(
+            user_id=request.user.id,
+            enterprise_customer=enterprise_customer,
+        )
+
+        if created:
+            enterprise_user.invite_key = enterprise_customer_key_match
+            enterprise_user.save()
+            track_enterprise_user_linked(
+                request.user.id,
+                pk,
+                enterprise_customer.uuid,
+            )
+        elif not enterprise_user.active or not enterprise_user.linked:
+            if not enterprise_user.active:
+                enterprise_user.active = True
+            if not enterprise_user.linked:
+                enterprise_user.linked = True
+            enterprise_user.save()
+
```

Comment:
You might want to return something other than a 201 in this `elif` block, since the enterprise user wasn't created - maybe a 200?

Also, I think you need an `else` path here to cover the case where the enterprise user already exists, is active, and linked to the enterprise.  And in that case you'd probably want to return a 204 response code.

Reply:
The blocks now return 200 codes. 

204 (`HTTP_204_NO_CONTENT`) would mean that our response wouldn't have the content `enterprise_customer_slug` and `enterprise_customer_uuid` which we would still need.